libs/ledger-live-common/src/families/near/api/archive-node-sdk.ts

```diff
@@ -0,0 +1,313 @@
+      stakingPositions,
+    },
+  };
+};
+
+export const getProtocolConfig = async (): Promise<NearProtocolConfig> => {
+  const { data } = await network({
+    method: "POST",
+    url: getEnv("API_NEAR_ARCHIVE_NODE"),
+    data: {
+      jsonrpc: "2.0",
+      id: "id",
+      method: "EXPERIMENTAL_protocol_config",
+      params: {
+        finality: "final",
+      },
+    },
+  });
+
+  return data.result;
+};
+
+export const getGasPrice = async (): Promise<string> => {
+  const { data } = await network({
+    method: "POST",
+    url: getEnv("API_NEAR_ARCHIVE_NODE"),
+    data: {
+      jsonrpc: "2.0",
+      id: "id",
+      method: "gas_price",
+      params: [null],
+    },
+  });
+
+  return data?.result?.gas_price;
+};
+
+export const getAccessKey = async ({
+  address,
+  publicKey,
+}: {
+  address: string;
+  publicKey: string;
+}): Promise<NearAccessKey> => {
+  const { data } = await network({
+    method: "POST",
+    url: getEnv("API_NEAR_ARCHIVE_NODE"),
+    data: {
+      jsonrpc: "2.0",
+      id: "id",
+      method: "query",
+      params: {
+        request_type: "view_access_key",
+        finality: "final",
+        account_id: address,
+        public_key: publicKey,
+      },
+    },
+  });
+
+  return data.result || {};
+};
+
+export const broadcastTransaction = async (
+  transaction: string
+): Promise<string> => {
+  const { data } = await network({
+    method: "POST",
+    url: getEnv("API_NEAR_ARCHIVE_NODE"),
+    data: {
+      jsonrpc: "2.0",
+      id: "id",
+      method: "broadcast_tx_async",
+      params: [transaction],
+    },
+  });
+
+  return data.result;
+};
+
+export const getStakingPositions = async (
+  address: string
+): Promise<{
+  stakingPositions: NearStakingPosition[];
+  totalStaked: BigNumber;
+  totalAvailable: BigNumber;
+  totalPending: BigNumber;
+}> => {
+  const stakingDeposits = await getStakingDeposits(address);
+
+  const { connect, keyStores } = nearAPI;
+
+  const config = {
+    networkId: "mainnet",
+    keyStore: new keyStores.InMemoryKeyStore(),
+    nodeUrl: getEnv("API_NEAR_ARCHIVE_NODE"),
+    headers: {},
+  };
+
+  const near = await connect(config);
+  const account = await near.account(address);
+
+  let totalStaked = new BigNumber(0);
+  let totalAvailable = new BigNumber(0);
+  let totalPending = new BigNumber(0);
+
+  const stakingPositions = await Promise.all(
+    stakingDeposits.map(async ({ validator_id: validatorId, deposit }) => {
+      const contract = new nearAPI.Contract(account, validatorId, {
+        viewMethods: [
+          "get_account_staked_balance",
+          "get_account_unstaked_balance",
+          "is_account_unstaked_balance_available",
+          "get_account_total_balance",
+        ],
+        changeMethods: [],
+      });
+
+      // Methods are dynamically added
+      /* eslint-disable @typescript-eslint/ban-ts-comment */
+      // @ts-ignore
+      const rawStaked = await contract.get_account_staked_balance({
```

Comment:
Is it possible to cast these method on a type like : 
```
type ContractParams {
account_id: string
}

type NearContract { 
get_account_staked_balance(params: ContractParam)
}
```
and then  assign `const contract  as NearContract` ?

I think you can probably bypass typescript
I don't have any other better way then this to remove all the //@ts-ignore

Reply:
Great idea, was able to implement this and remove the `//@ts-ignore`